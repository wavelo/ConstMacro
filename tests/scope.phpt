<?php

include 'bootstrap.php';


$props = [0,1,2];
$a = 10;
$b = 3;
$parser = ConstMacro\Parser::parse("[a,b,c] = \$props");

Assert::equal(10, $a);
Assert::equal(3, $b);
Assert::false(isset($c));

$compacted = call_user_func_array('compact', $parser->compact) + array_fill_keys($parser->compact, NULL);

eval($parser->expr);

Assert::equal(0, $a);
Assert::equal(1, $b);
Assert::true(isset($c));
Assert::equal(2, $c);

extract($compacted);

Assert::equal(10, $a);
Assert::equal(3, $b);
Assert::false(isset($c));
