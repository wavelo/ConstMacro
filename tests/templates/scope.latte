
{default $title=>'title #0'}
{default $props1=>['title'=>'title #1']}
{default $props2=>['title'=>'title #2']}
{default $props3=>[]}

<div>
	<p>{$title}</p>

	{const [title:] = $props1}
	<p>{$title}</p>
	{/const}
</div>

<div>
	<p>{$title}</p>

	<div n:const="[title:] = $props1">
		<p>{$title}</p>

		{var title=>'title #3'}

		<div>
			{const [title:] = $props2}
			<p>{$title}</p>
			{/const}
		</div>

		<p>{$title}</p>
	</div>
</div>

<div>
	<p>{$title}</p>

	{const [title:] = $props1 /}
	<p>{$title}</p>
</div>

<div>
	<p>{$title}</p>

	<div>
		{const [title:="title #4"] = $props3}
		<p>{$title}</p>
		{/const}
	</div>
</div>
