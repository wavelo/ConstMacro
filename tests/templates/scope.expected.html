<div>
	<p>title #0</p>
	<p>title #1</p>
</div>

<div>
	<p>title #0</p>

	<div>
		<p>title #1</p>

		<div>
			<p>title #2</p>
		</div>

		<p>title #3</p>
	</div>
</div>

<div>
	<p>title #0</p>
	<p>title #1</p>
</div>

<div>
	<p>title #1</p>

	<div>
		<p>title #4</p>
	</div>
</div>
